lib/carthage_cache/application.rb

```diff
@@ -16,8 +16,13 @@ def initialize(project_path, verbose, config, repository: AWSRepository, termina
       @terminal = terminal.new(verbose)
       @archiver = Archiver.new
       @config = Configurator.new(@terminal, project_path, config).config
-      clazz = @config.read_only? ? HTTPRepository : repository
-      @repository = clazz.new(@config.bucket_name, @config.hash_object[:aws_s3_client_options])
+      if @config.local_only?
+          cacheDirectory = File.join(project_path, "Carthage", @config.local_mode)
```

Comment:
Yeah I don't want to share the cache across projects. 
Reason for this was that if I need to clean out the caches it means I'll have to remove all of the cache as the zips themselves aren't identifiable to which repo.
The directory check happens inside of LocalRepository initialize